File: Hotel-management/Hotel-management/src/main/java/com/example/Hotel_management/dto/BookingDTO.java
package com.example.Hotel_management.dto;

import com.example.Hotel_management.entity.Room;
import com.example.Hotel_management.user.User;
import com.fasterxml.jackson.annotation.JsonInclude;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.validation.constraints.Future;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.Builder;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Builder
@JsonInclude(JsonInclude.Include.NON_NULL)
public class BookingDTO {
    private Integer Id;

    private LocalDate checkInDate;


    private LocalDate checkOutDate;


    private int numOfAdults;


    private int numOfChildren;

    private int totalNumOfGuest;


    private String bookingConfigurationCode;

    private UserDTO user;

    private RoomDTO room;
}

File: hotel-management-ui/src/app/services/models/booking-dto.ts
/* tslint:disable */
/* eslint-disable */
import { RoomDto } from '../models/room-dto';
import { UserDto } from '../models/user-dto';
export interface BookingDto {
  bookingConfigurationCode?: string;
  checkInDate?: string;
  checkOutDate?: string;
  id?: number;
  numOfAdults?: number;
  numOfChildren?: number;
  room?: RoomDto;
  totalNumOfGuest?: number;
  user?: UserDto;
}
